### PR TITLE
Add code to not cleanup the server if there are other connections to it.

### PR DESCRIPTION
**Pull request by dwighthubbard**
Added code to only run cleanup if the server connection count == 1 so the redis server will not terminate if it's still in use.

Add test to verify the server doesn't go away if the first redislite instance pointing to it goes away.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: redislite/client.py [hunk 1]
```diff
@@ -67,36 +67,64 @@ def _cleanup(self):
         :return:
         """
 
-        if not self.redis_dir:  # pragma: no cover
-            return
-
-        if self.running:
+        #if not self.redis_dir:  # pragma: no cover
```
Comment (slupers) [inline redislite/client.py]: remove commented code?

File: redislite/client.py [hunk 2]
```diff
@@ -326,7 +360,16 @@ def pid(self):
...
             with open(self.pidfile) as fh:
-                pid = fh.read().strip()
+                pid = int(fh.read().strip())
+                if pid:  # pragma: no cover
+                    try:
+                        p = psutil.Process(pid)
+                    except psutil.NoSuchProcess:
+                        return 0
+                    if not p.is_running():
+                        return pid
```
Comment (slupers) [inline redislite/client.py]: should be return 0 if not running?

File: tests/test_unit.py [hunk 3]
```diff
@@ -18,10 +21,53 @@
...
+            args=tuple(new_args)
+            return logger.debug(*args, **kwargs)
+
+    def _redis_server_processes(self):
+        """
+        Return count of process objects for all redis-server processes started
+        by the current user.
+        :return:
+        """
+        import psutil
```
Comment (slupers) [inline tests/test_unit.py]: already imported at the top

File: tests/test_unit.py [hunk 4]
```diff
@@ -271,10 +356,63 @@ def test_metadata(self):
...
+        )
+        self.logger.debug(
+            'Second connection count is: %s', s._connection_count()
+        )
+        s._cleanup()
+        with self.assertRaises(psutil.NoSuchProcess):
+            p = psutil.Process(pid)
+
+
+    def test_connection_count(self):
```
Comment (slupers) [inline tests/test_unit.py]: maybe test multiple connection count as well

**Pull request merged**